src/core/column/mean.h

```diff
@@ -25,20 +25,20 @@
 namespace dt {
 
 
-template <typename T, bool IS_GROUPED>
-class Mean_ColumnImpl : public ReduceUnary_ColumnImpl<T, IS_GROUPED> {
+template <typename T_IN, typename T_OUT, bool IS_GROUPED>
+class Mean_ColumnImpl : public ReduceUnary_ColumnImpl<T_IN, T_OUT, IS_GROUPED> {
```

Comment:
For mean, for instance, you only need to pass one `T` and then call `ReduceUnary_ColumnImpl<T, T, IS_GROUPED>` — because you already casted the input column to the corresponding stype. May be I don't really understand what you're trying to accomplish here...